lib/phoenix_live_view/html_formatter.ex

```diff
@@ -391,7 +393,17 @@ defmodule Phoenix.LiveView.HTMLFormatter do
   end
 
   defp to_tree([{:tag_close, name, _meta} | tokens], buffer, [{name, attrs, upper_buffer} | stack]) do
-    tag_block = {:tag_block, name, attrs, Enum.reverse(buffer)}
+    format? =
+      case List.first(upper_buffer) do
+        {:text, text, _meta} ->
+          :binary.last(text) in '\s'
+
+        _ ->
+          true
+      end
+
+    tag_block = {:tag_block, name, attrs, Enum.reverse(buffer), %{format?: format?}}
```

Comment:
@josevalim please, take a look and let me know if this solution make sense. My goal was to try to identify if a tag is surrounded by texts while building the tree and, then adding a metadata that this tag should not be formatted. Finally, in the HTML algebra we check if the element is inline and for this metadata. 

Let me know if this solution make sense at all so I can improve the code + add some docs.

Reply:
Perfect!